docs/community/tutorials-contest/2022/red-dev-ecrecover-x-chain/README.md

```diff
@@ -0,0 +1,362 @@
+}
+```
+
+Fourth, at about [line 533](https://github.com/ava-labs/subnet-evm/blob/master/params/config.go#L533), set the enabled status of the `IsContractXChainECRecoverEnabled` by checking whether a fork scheduled at given block timestamp is active at the given head block:
+
+```diff
+func (c *ChainConfig) AvalancheRules(blockNum, blockTimestamp *big.Int) Rules {
+  .....
+  .....
+  rules.IsContractDeployerAllowListEnabled = c.IsContractDeployerAllowList(blockTimestamp)
+	rules.IsContractNativeMinterEnabled = c.IsContractNativeMinter(blockTimestamp)
+	rules.IsTxAllowListEnabled = c.IsTxAllowList(blockTimestamp)
++ 	rules.IsContractXChainECRecoverEnabled = c.IsXChainECRecover(blockTimestamp)
+
+	// Initialize the stateful precompiles that should be enabled at [blockTimestamp].
+	rules.Precompiles = make(map[common.Address]precompile.StatefulPrecompiledContract)
+    .....
+
+	return rules
+}
+```
+
+And fifth, at about [line 563](https://github.com/ava-labs/subnet-evm/blob/master/params/config.go#L563),  include your custom precompiles into the stateful precompiled configs:
+
+```diff
+func (c *ChainConfig) enabledStatefulPrecompiles() []precompile.StatefulPrecompileConfig {
+	statefulPrecompileConfigs := make([]precompile.StatefulPrecompileConfig, 0)
+
+	if c.ContractDeployerAllowListConfig.Timestamp() != nil {
+		statefulPrecompileConfigs = append(statefulPrecompileConfigs, &c.ContractDeployerAllowListConfig)
+	}
+    	.....
+    	.....
++   	if c.ContractXChainECRecoverConfig.Timestamp() != nil {
++ 		statefulPrecompileConfigs = append(statefulPrecompileConfigs, &c.ContractXChainECRecoverConfig)
++ 	}
+}
+```
+### Modify the Run Script
+
+The script [scripts/run.sh](https://github.com/ava-labs/subnet-evm/blob/master/scripts/run.sh) by default includes the default configuration. Since, the precompiles created under precompile folder are optional and can be added to the Subnet-evm by configuring in the `genesis.json` file.
+
+Modify the default `genesis.json` setting in the [scripts/run.sh](https://github.com/ava-labs/subnet-evm/blob/master/scripts/run.sh) to enable the custom precompile. At about [line 125](https://github.com/ava-labs/subnet-evm/blob/master/scripts/run.sh#L125) under the `config` object, add the configuration for custom precompile. The configuration name `contractXChainECRecover` can be derived from [params/config.go line 122](https://github.com/ava-labs/subnet-evm/blob/master/params/config.go#L122):
+
+```diff
+{
+  "config": {
+    "chainId": $CHAIN_ID,
+    "homesteadBlock": 0,
+    "eip150Block": 0,
+    .....
+    .....
+    "feeConfig": {
+      "gasLimit": 20000000,
+      .....
+      .....
+      "blockGasCostStep": 500000
+    },
++     "contractXChainECRecover": {
++       "blockTimestamp": 0
++     }
+  },
+  "alloc": {
+    "${GENESIS_ADDRESS:2}": {
+      "balance": "0x52B7D2DCC80CD2E4000000"
+    }
+  },
+  "nonce": "0x0",
+  ....
+  ....
+
+}
+```
+
+## Run the Local Subnet-evm
```

Comment:
```suggestion
## Run the Local Subnet EVM
```

Reply:
@Julian-dev28 We have made the requested changes as per your suggestions. Please let us know if you need anything more.

Reply:
You guys have been great

Thanks for the hard work!